decorators.py

```diff
@@ -2,14 +2,16 @@
 from django.db.models import Q
 from django.http import JsonResponse
 
-from api import views as api_views
-
-# TODO: might want something similar so people can only write to their own data in some models
-# this only deals with getting things back since all writing requires login so it not open
+from api.search_helpers import get_query_tuple
 
 
 def apply_model_get_restrictions(function):
-    """Apply model restrictions to the data returned by the API."""
+    """Apply model restrictions to the data returned by the API.
+
+    This only deals with getting things back since all writing requires login so it not open.
```

Comment:
Not sure if this line is complete, suggest ->
```suggestion
    "This decorator handles read-only access, as all write operations require authentication. It ensures that data retrieval respects model-specific availability settings."
```